pure/Data/Aeson/Parser/UnescapePure.hs

```diff
@@ -172,11 +170,11 @@ unescapeText' bs = runText $ \done -> do
         (st', p) ->
             return (pos, StateUtf st' p)
 
-      {-# INLINE runUtf #-}
-
-      f' dest m c = m >>= \s -> f dest s c
-
-      {-# INLINE f' #-}
+      loop _ ps i | i >= len = return ps
+      loop dest ps i = do
+        let c = B.index bs i -- JP: We can use unsafe index once we prove bounds with Liquid Haskell.
```

Comment:
Access fine because of line 173 wit definition of len on line 160.
i is checked to smaller than the len in 173, so B.index should not throw.

Also bs is not an argument to loop (I just confused it with ps) - so it never changes.